src/components/ui/select.tsx

```diff
@@ -1,58 +1,166 @@
 'use client';
 
-import type { CollectionItem } from '@chakra-ui/react';
-import { Select as ChakraSelect, Portal } from '@chakra-ui/react';
+import { CurvedCornerIcon } from '@/ui/icons/CurvedCornerIcon';
+import { selectSlotRecipe } from '@/ui/theme/recipes/SelectRecipe';
+import type { CollectionItem, RecipeVariantProps } from '@chakra-ui/react';
+import { Box, Select as ChakraSelect, Flex, Icon, Portal } from '@chakra-ui/react';
 import * as React from 'react';
 
-import { CloseButton } from './close-button';
+export const CURVED_CORNER_SIZE = 4;
 
+interface SelectControlProps extends ChakraSelect.ControlProps {
+  open: boolean;
+  isContentWiderThanControl?: boolean;
+  positioning?: ChakraSelect.RootProps['positioning'];
+}
+export const SelectControl = React.forwardRef<HTMLDivElement, SelectControlProps>(
+  function SelectControl(props, ref) {
+    const { open, isContentWiderThanControl, positioning, ...rest } = props;
+    const placement = positioning?.placement ?? 'bottom-start';
+    const mainAxis = positioning?.offset?.mainAxis ?? 0;
+
+    return (
+      <Box position="relative">
+        <ChakraSelect.Control
+          {...rest}
+          ref={ref}
+          borderBottomLeftRadius={open ? 'none' : 'redesign.lg'}
+          borderBottomRightRadius={open ? 'none' : 'redesign.lg'}
```

Comment:
these are already defined in recipe?

Reply:
Dyanmic styles that rely on state cannot be put in recipes